### DISCUSSION TITLE
Python Org babel Async execution

### DISCUSSION THREAD

**Discussion by Geegee** [syl20bnr/spacemacs#16722] [Q&A] [answered]
The current behavior is that if I run a source code block in **org babel python**
emacs completely hangs. I am not sure if this is the default behavior or
some misconfiguration on my side so I would like to hear if someone
knows more in this regard. Maybe I could also send the blocks to another
terminal or shell process might also be a work around for me.

**Comment by fnussbaum** [comment, marked as answer]
By default, the babel blocks run synchronously. The following should run asynchronously:
```python
#+begin_src python :session :async
import time
time.sleep(5)
42
#+end_src
```
